src/battle/battle_controller_player.c

```diff
@@ -84,10 +85,563 @@ void BattleControllerPlayer_PokemonAppear(BattleSystem *bsys, BATTLECONTEXT *ctx
+                BattleController_EmitDrawYesNoBox(bsys, ctx, battlerId, 955, 0, 0, 0);
+                ctx->unk_0[battlerId] = 12;
+            } else if ((battleType & BATTLE_TYPE_SINGLES) && !(battleType & BATTLE_TYPE_LINK)) {
+                if (BattleSystem_GetBattleFlags(bsys) & 0x10) {
+                    ov12_0223BFFC(bsys, 1);
+                    ov12_022581BC(bsys, ov12_0223A7E4(bsys));
+                } else {
+                    msg.tag = 0;
+                    msg.id = 793; //TODO: replace with msg constants
+                    ov12_022639B8(bsys, battlerId, msg);
+                    ctx->unk_0[battlerId] = 15;
+                    ctx->unk_4[battlerId] = 0;
+                }
+            } else if (CantEscape(bsys, ctx, battlerId, &msg)) {
+                if (BattleSystem_GetBattleFlags(bsys) & 0x10) {
+                    ov12_0223BFFC(bsys, 1);
+                    ov12_022581BC(bsys, ov12_0223A7E4(bsys));
+                } else {
+                    ov12_022639B8(bsys, battlerId, msg);
+                    ctx->unk_0[battlerId] = 15;
+                    ctx->unk_4[battlerId] = 0;
+                }
+            } else {
+                ctx->unk_0[battlerId] = 12;
+            }
+            break;
+        case 12:
+            if (battleType & BATTLE_TYPE_TOWER) {
+                if (ov12_0225682C(ctx, battlerId)) {
+                    if (ov12_0225682C(ctx, battlerId) == 0xff) {
+                        ctx->unk_0[battlerId] = 0;
+                    } else {
+                        ctx->unk_0[battlerId] = 13;
+                    }
+                }
+            } else {
+                ctx->unk_0[battlerId] = 13;
+            }
+
+            if ((battleType & BATTLE_TYPE_DOUBLES) && !(battleType & BATTLE_TYPE_MULTI) && (ctx->unk_0[battlerId] == 13)) {
+                ctx->unk_0[BattleSystem_GetBattlerIdPartner(bsys, battlerId)] = 13;
+            }
+            break;
+        case 13: //WIFI wait for partner to make a move..?
+            ov12_02263CCC(bsys, battlerId);
+
+            if (battleType == (BATTLE_TYPE_SINGLES | BATTLE_TYPE_DOUBLES | BATTLE_TYPE_LINK)) {
+                int partnerId;
+
+                partnerId = BattleSystem_GetBattlerIdPartner(bsys, battlerId);
+
+                if (ctx->unk_0[partnerId] == 14) {
+                    BattleController_EmitShowWaitMessage(bsys, battlerId);
+                }
+            } else {
+                BattleController_EmitShowWaitMessage(bsys, battlerId);
+            }
+
+            ctx->unk_0[battlerId] = 14;
+        case 14:
+            var++; //this might be counting the number of players who have selected their final action?
+            break;
+        case 15:
+            if (ov12_0225682C(ctx, battlerId)) {
+                ov12_0224EDC0(ctx, battlerId);
+                ctx->unk_0[battlerId] = ctx->unk_4[battlerId];
+            }
+            break;
+        case 16:
+            msg.tag = 2;
+            msg.id = 608;
```

Comment:
constant

Reply:
where are the battle message constants again?

Reply:
nvm I found them